added institute information in variable

valeriupredoi opened the pull request:
@bouweandela in `_data_finder.py`: `var` for fx_files and `variable` for regular files needs the `institute` key set, otherwise line 116 from `_data_finder.py` will kick in the error that `tag` is orphan - are you setting the institute key anywhere else? DRS: BADC will not work if `institute` is not a variable key.

Closes #665.

mattiarighi commented:
I've tested this with DKRZ and ETHZ drs.

valeriupredoi commented:
also, my tests are failing on a local machine:
```
tests/system/data_simulator.py:10: in <module>
    from dummydata.model2 import Model2
E   ImportError: No module named dummydata.model2
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================================================== 1 error in 7.20 seconds ========================
```
what's this dummydata crap? :grin:

bouweandela commented:
> If you can explain this

You probably made a mistake at some point when merging branches. If you look at this https://github.com/ESMValGroup/ESMValTool/pull/655 you can see that it overwrites the required lines of code..

> what's this dummydata crap

This is the dummydata module https://github.com/ESMValGroup/dummydata
we have an open issue on it #447 
Probably something went wrong with your installation, it's installed from environment.yml when you do conda update --file environment.yml --name your_esmvaltool_environment as described in README.md

valeriupredoi commented:
@bouweandela this is a bit less trivial than just a mistake - I have just repeated the merge:
```
valeriu@valeriu-PORTEGE-Z30-C:~/esmvaltool_v2dev$ git status
On branch version2_development_autoassess_without_conservation
nothing to commit, working directory clean
valeriu@valeriu-PORTEGE-Z30-C:~/esmvaltool_v2dev$ git pull origin version2_development_autoassess_without_conservation 
From https://github.com/ESMValGroup/ESMValTool
 * branch            version2_development_autoassess_without_conservation -> FETCH_HEAD
Already up-to-date.
valeriu@valeriu-PORTEGE-Z30-C:~/esmvaltool_v2dev$ git checkout version2_development
Switched to branch 'version2_development'
Your branch is up-to-date with 'origin/version2_development'.
valeriu@valeriu-PORTEGE-Z30-C:~/esmvaltool_v2dev$ git pull origin version2_development
remote: Enumerating objects: 7, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 7 (delta 3), reused 7 (delta 3), pack-reused 0
Unpacking objects: 100% (7/7), done.
From https://github.com/ESMValGroup/ESMValTool
 * branch            version2_development -> FETCH_HEAD
   530d886..6a29baa  version2_development -> origin/version2_development
Updating 530d886..6a29baa
Fast-forward
 doc/sphinx/source/figures/schematic.png | Bin 281988 -> 269597 bytes
 1 file changed, 0 insertions(+), 0 deletions(-)
valeriu@valeriu-PORTEGE-Z30-C:~/esmvaltool_v2dev$ git checkout version2_development_autoassess_without_conservation 
Switched to branch 'version2_development_autoassess_without_conservation'
valeriu@valeriu-PORTEGE-Z30-C:~/esmvaltool_v2dev$ git merge version2_development
Merge made by the 'recursive' strategy.
 doc/sphinx/source/figures/schematic.png | Bin 281988 -> 269597 bytes
 1 file changed, 0 insertions(+), 0 deletions(-)
```
and `_recipe.py` is still broken

valeriupredoi commented:
@bouweandela thoroughly confused since I didnt make any mistake and even repeated the merge from a clean git repo, but who knows, shit must have hit some fan somewhere. Anyway, I just realigned `_recipe.py` in #655 and all should be good now, am gonna cancel the PR and delete the branch :confused:

mattiarighi commented:
It works now! 
Thanks for fixing this. :+1:

schlunma commented:
Can I merge this?

mattiarighi commented:
Let's wait for @valeriupredoi to test it.

valeriupredoi commented:
wowza guys, good stuff! I been busy with the reformatting workflow, gonna test in a smoke's time

valeriupredoi commented:
yep it works fine with BADC (model and fx files as well) :balloon:

bouweandela commented:
The bug that @schlunma fixed in this PR was introduced in PR #632, while the bug that @valeriupredoi fixed did not exist at all in the `version2_development` branch, as extensively discussed above.
